### PR TITLE
TST: Split test_offsets.py

### LINKED ISSUE #27085: TST: Split test_offsets.py
xref https://github.com/pandas-dev/pandas/pull/26628#issuecomment-500166805

It would be good to create:

`pandas/tests/tseries/offsets/test_business_offsets.py` for business offsets
`pandas/tests/tseries/offsets/test_date_offsets.py` for all offsets that subclass `DateOffset`

and converting the tests to pytest idioms.

### PR DESCRIPTION
**Pull request by Raalsky**
- [x] closes #27085
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/tseries/offsets/conftest.py [hunk 1]
```diff
@@ -2,9 +2,59 @@
...
+    """
+    Fixture for all the datetime offsets available for a time series.
+    """
+    return request.param
+
+
+@pytest.fixture(params=[getattr(offsets, o) for o in BUSINESS_OFFSETS])
+def business_offset_types(request):
     """
     Fixture for all the datetime offsets available for a time series.
```
Comment (simonjayhawkins) [inline pandas/tests/tseries/offsets/conftest.py]: same docstring for business_offset_types as date_offset_types docstring. should differentiate.

File: pandas/tests/tseries/offsets/conftest.py [hunk 2]
```diff
@@ -14,11 +64,25 @@ def offset_types(request):
...
+    params=[
+        getattr(offsets, o)
+        for o in BUSINESS_OFFSETS
         if issubclass(getattr(offsets, o), offsets.MonthOffset) and o != "MonthOffset"
     ]
 )
-def month_classes(request):
+def business_month_classes(request):
     """
     Fixture for month based datetime offsets available for a time series.
```
Comment (simonjayhawkins) [inline pandas/tests/tseries/offsets/conftest.py]: same docstring for business_month_classes as date_month_classes. should differentiate.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @Raalsky for the PR. making changes and splitting files in the same PR make review more difficult. Is is possible to do just the split or just the fixturisation in this PR and then a follow-on?

**Review by jreback — CHANGES_REQUESTED** [review]
yeah if you can do a straight move *first*, then we can fixturize (or the other way around is ok too). having multiple targeted PRs is likely to make review / merge much faster.

**Comment by jbrockmendel** [comment]
@Raalsky can you rebase

**Comment by jreback** [comment]
closing as stale. @Raalsky would love a PR that first moves, then followups to split / parameterize this. make small targeted PRs; we can then merge quickly.

**Comment by Raalsky** [comment]
Sorry for being off during the holidays. I've prepared the first step (move) test_offsets to test_date_offsets [31031](https://github.com/pandas-dev/pandas/pull/31031) Will be following by adding fixtures and then business/date split.

**Comment by Raalsky** [comment]
Fixtures added in #31034

File: pandas/tests/tseries/offsets/conftest.py (hunk 2 above)
Comment (Raalsky) [reply pandas/tests/tseries/offsets/conftest.py]: Added in #31034

File: pandas/tests/tseries/offsets/conftest.py (hunk 1 above)
Comment (Raalsky) [reply pandas/tests/tseries/offsets/conftest.py]: Added in #31034

**Pull request closed without merging**